components/pixels/lib.rs

```diff
@@ -376,8 +395,11 @@ fn decode_gif(buffer: &[u8], cors_status: CorsStatus) -> Option<Image> {
             };
             rgba8_byte_swap_colors_inplace(frame.buffer_mut());
 
+            let frame_start = bytes.len();
+            bytes.extend_from_slice(&frame.buffer());
+
```

Comment:
It's a bit inefficient to keep extending the bytes array. Can you collect a collect an array of bytes and then join them all, which should only do a single extra allocation?